src/core_plugins/kibana/public/dashboard/dashboard.js

```diff
@@ -301,12 +301,17 @@ app.directive('dashboardApp', function (Notifier, courier, AppState, timefilter,
           `${VisualizeConstants.WIZARD_STEP_1_PAGE_PATH}?${DashboardConstants.ADD_VISUALIZATION_TO_DASHBOARD_MODE_PARAM}`);
       };
 
+      const addNewSearch = function addNewSearch() {
+        kbnUrl.change('/discover');
+      };
```

Comment:
I really like the intention behind this. We should definitely make it as easy to add a new Saved Search as it is to add a new Visualization. The problem is that the logic for adding a new visualization from a dashboard uses a special query parameter and additional routing logic to ensure that you're routed back to the dashboard you came from, once the new visualization is saved. Here's the entry point into this logic if you're curious: https://github.com/elastic/kibana/blob/master/src/core_plugins/kibana/public/dashboard/dashboard.js#L301.

When we implement a similar feature here, we should follow the same pattern and code style, and provide the same user experience. However, I think those changes are beyond the scope of this PR, and are best reviewed in a subsequent, separate PR.

So for now, would you mind removing the changes you made that allow users to add a new saved search from a dashboard? I'd be happy to review these changes in a separate PR.